x/commercioid/querier.go

```diff
@@ -48,9 +49,13 @@ func queryResolveIdentity(ctx sdk.Context, path []string, keeper Keeper) (res []
 	identityResult.Did = did
 	identityResult.DdoReference = keeper.GetDdoReferenceByDid(ctx, did)
 
+	if identityResult.DdoReference == "" {
+		return nil, sdk.ErrUnknownRequest(fmt.Sprintf("No ddo reference related to given did"))
+	}
+
 	bz, err2 := codec.MarshalJSONIndent(keeper.cdc, identityResult)
 	if err2 != nil {
-		panic("Could not marshal result to JSON")
+		return nil, sdk.ErrUnknownRequest(fmt.Sprintf("Could not marshal result to JSON"))
```

Comment:
If we do not have any error variables I think we can remove the usage of `fmt.Sprintf` and leave the plain string.